src/broadcast/spectate_manager.ts

```diff
@@ -304,4 +304,25 @@ export class SpectateManager extends EventEmitter {
       dolphinId: value.dolphinId,
     }));
   }
+
+  public async disconnect(): Promise<void> {
+    this.emit(SpectateEvent.BROADCAST_LIST_UPDATE, []);
+    this.emit(SpectateEvent.LOG, "Disconnecting from spectate server");
+
+    // Stop watching all broadcasts
```

Comment:
since this code path is currently only used to disconnect while idle, should we check `openBroadcasts` first and not continue if there are indeed open broadcasts?